toolkit/tools/scheduler/scheduler.go

```diff
@@ -277,6 +277,26 @@ func startWorkerPool(agent buildagents.BuildAgent, workers, buildAttempts, check
 	return
 }
 
+// debugStuckNode is a debugging function that will print out the stuck node and all nodes that are blocking it.
+func debugStuckNode(buildState *schedulerutils.GraphBuildState, pkgGraph *pkggraph.PkgGraph, stuckNode *pkggraph.PkgNode, indent int) {
```

Comment:
I've noticed this right now: at the end of each successful package build we run `graphanalytics`, which already prints a nice summary of what got blocked. I'm curious, if instead of adding this function, we could run the logic from that one but regardless if a build failure or succeeded. Either directly from the scheduler or through modifying our make files.
![image](https://github.com/microsoft/CBL-Mariner/assets/24633722/ddc65fa7-d468-4824-bf47-632197103080)

Reply:
At this point I think it will be hard to reconstruct the failure state. The analytics tool only considers the node state/type (ie unresolved remote nodes). As best I can tell the failures we are seeing are in the actual scheduler state rather than the graph.